app/models/game.rb

```diff
@@ -87,6 +87,108 @@ def is_in_check?(player)
     return false
   end
 
+  def is_in_checkmate?(player)
+    return false if !is_in_check?(player)
+    if player == white_player
```

Comment:
Ideally the logic would be written so that it doesn't matter which player color you are checking.